src/Database/Driver/OracleBase.php

```diff
@@ -377,25 +382,37 @@ public function prepareMethod($queryString, $options = [])
      *
      * @param string $table Table name.
      * @param string $column Column name
+     * @param string $schema Schema name
      * @return int
      */
-    protected function _autoincrementSequenceId(?string $table, ?string $column)
+    protected function _autoincrementSequenceId(?string $table, ?string $column, ?string $schema)
     {
+        $schemaExists = isset($schema);
```

Comment:
The introduction of this `$schemaExists` flag raises some maintainability risk.  Conditional logic is performed in the `if ($this->isAutoQuotingEnabled())` statement which may alter the `$schema` variable.  If a future change looks up the schema in the `if` or `else` condition, it would be responsible for updating the `$schemaExists` flag.

The existing pattern appears to be: take in option parameters as arguments (`$table`, `$column`), then assign them to trusted variables (`$tableName`, `$columnName`) which can be used in the SQL queries.  Instead of introducing a new flag variable, follow that convention to created a trusted `$schemaName` variable, which is check via `isset()` as needed.

Reply:
Would it be better to just move it to after the conditional logic. While they do use isset or empty within if statements. Others like the `if ($this->isAutoQuotingEnabled())` is just returning a boolean, and in the construct statement of OCI8Connection they do a very similar thing with setting `$persistent = !empty($options['persistent']);` and then a little bit down checking `if ($persistent)`. 

They also don't have much consistency because the very next line of code is `if ($charset !== null)` and the charset was set right after persistent with `$charset = !empty($options['charset']) ? $options['charset'] : null;`.  So, I'm not sure which would be better. I was just trying to make sure the code wouldn't be repeating itself for a simple check.

Reply:
The convenience variables of `$persistent`, `$charset`, etc. in `OCI8Connection::_construct()` take a complex logic statement and condense it to a single readable variable.  The function also treats these as immutable; it has no incentive or business modifying the `$options[]` array.

In the examples here, the arguments are deliberately modified: optional, `strtoupper`, etc.  Introducing the convenience variable introduces the cognitive load of also keeping it current as changes occur (or enforcing that it is only set after the last change has occurred).

I agree that there isn't a ton of internal consistency here, but adding the convenience variable is also a clear move away from the immediate consistency of the surrounding code, without any real wins in aligning it with broader consistency across other files.

One other tangential thought: is there precedent for preferring:
```
        if ($schemaExists) {
            $query = "select sequence_name from all_tab_identity_cols where table_name='$tableName' and column_name='$columnName' and owner='$schema'";
        } else {
            $query = "select sequence_name from all_tab_identity_cols where table_name='$tableName' and column_name='$columnName'";
        }
```
over
```
        $query = "select sequence_name from all_tab_identity_cols where table_name='$tableName' and column_name='$columnName'";
        if ($schemaExists) {
            $query. =  and owner='$schema'";
        }
```

The latter reduces duplicate code, but some projects may prefer duplication in favor of more readable SQL.

Reply:
I understand what you are saying, I will change it to use the isset since their is no clear advantage. 

Additionally no, I had no preference. I was just doing it that way since the next time I use schema, it wasn't easily broken apart here
```php
if ($schemaExists) {
      $statement = $this->_connection->query("SELECT {$schemaName}.{$sequenceName}.CURRVAL FROM DUAL");
} else {
      $statement = $this->_connection->query("SELECT {$sequenceName}.CURRVAL FROM DUAL");
}
```
Additionally, the getDSN function combines multiple strings together 
```php
return '(DESCRIPTION=(ADDRESS=(PROTOCOL=TCP)(HOST=' . $config['host'] . ')(PORT=' . $config['port'] . '))' . '(CONNECT_DATA=(' . $service . ')' . $instance . $pooled . '))';```
So, I think I will break it out like in your suggestion. This leads to another question, should my second if statement be changed to be similar to the first. As in, 
```php
$query = "SELECT ";
if (isset($schemaName)) {
     $query .= "{$schemaName}.";
}
$query .= "{$sequenceName}.CURRVAL FROM DUAL";
$statement = $this->_connection->query($query);
```. This is how the query that you referenced works, so I'm not sure why it was changed for the second one.

Reply:
I'd default a ternary, personally
```
"SELECT ".(isset($schemaName) ? "{$schemaName}." : "")."{$sequenceName}.CURRVAL FROM DUAL"
```